build/build-image/Dockerfile

```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # This file creates a standard build environment for building Kubernetes
-FROM k8s.gcr.io/kube-cross:KUBE_BUILD_IMAGE_CROSS_TAG
+FROM us.gcr.io/k8s-artifacts-prod/build-image/kube-cross:KUBE_BUILD_IMAGE_CROSS_TAG
```

Comment:
`k8s-artifacts-prod/build-image/kube-cross` seems a bit long, no? wanna do `k8s-artifacts-prod/kube-cross`

Reply:
@dims -- Yes, it's definitely longer that I'd like :joy: 
I was just following the convention here, but @thockin was mentioning that there could be exceptions made for certain images. This might qualify as one of them.

Reply:
I don't reeeeeally care about length for this image. `pause` might be an exception, but not this one I don't think. /shrug

Reply:
why the geo-centric change? I don't know enough to know what this does

Reply:
https://cloud.google.com/container-registry/docs/pushing-and-pulling#tag_the_local_image_with_the_registry_name

The current gcr.io is US based. When we move k8s.gcr.io over I believe it will be global then.

@listx or @thockin can probably elaborate

Reply:
Yep, yep, what @cblecker said (from my understanding).